### PR TITLE
DOC: normalize usage of word "pandas"

### LINKED ISSUE #32316: DOC: Standardize references to pandas in the documentation
In our documentation (files in `doc/source/../*.rst`), we refer to pandas as `Pandas`, `pandas`, and `*pandas*` (sphinx italics).

For documentation consistency, we should standardize these to one form. `pandas` is the preferable form in our documentation files.

Unfortunately I don't think we create a lint rule for this pattern since we have to be flexible to references and hyperlinks.

### PR DESCRIPTION
**Pull request by LeviMatus**
changes references to the library, pandas, to match the standard lowercase spelling. This changes applicable `.rst` files under the `doc/source` path.

References such as Pandas, **pandas**, *pandas*, and ``pandas`` have been replaced with pandas.

- [x] closes #32316 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.2.0.rst [hunk 1]
```diff
@@ -287,6 +287,12 @@ Performance improvements
 
 .. ---------------------------------------------------------------------------
 
+.. _whatsnew_0230.docs:
+
+Documentation changes
+~~~~~~~~~~~~~~~~~~~~~
+- Replaced tokens Pandas, **pandas**, *pandas*, and ``pandas`` with pandas for consistency (:issue:`32316`).
+
```
Comment (dsaxton) [inline doc/source/whatsnew/v1.2.0.rst:295]: I don't think we'd need a release note for this

**Review by dsaxton — COMMENTED** [review]
Thanks @LeviMatus, does this fix all remaining references to pandas in the docs? If not we don't want to close the original issue.

**Comment by LeviMatus** [comment]
@dsaxton, I combed through all of `doc/source/../*.rst` which is what #32316 was referring to. I'm fairly certain this catches all remaining references.

**Review by mroeschke — APPROVED** [review]
Interested in writing a doc linting rule such that we catch future usages of `Pandas`, etc?

(Can be a follow up PR)

**Comment by LeviMatus** [comment]
> Interested in writing a doc linting rule such that we catch future usages of `Pandas`, etc?

I can take some time to look into doing that.

**Comment by dsaxton** [comment]
Thanks again @LeviMatus, feel free to open a PR with the linting check as a follow-up

**Pull request merged**